exercises/practice/complex-numbers/generate_tests.exs

```diff
@@ -0,0 +1,158 @@
+# Generates test cases and basic template, which will need to be adapted by hand
+# Pass the name of the exercise (e. g, "complex-numbers") as an argument
+
+defmodule Generate do
+  def to_snake_case(lower_camel_case) do
+    lower_camel_case
+    |> String.split(~r/(?=[A-Z\d])/)
+    |> Enum.map_join("_", &String.downcase/1)
+  end
+
+  def get_properties(%{"cases" => cases}) when is_list(cases) do
+    Enum.map(cases, &Generate.get_properties/1)
+    |> Enum.concat()
+    |> Enum.uniq()
+  end
+
+  def get_properties(%{"property" => property, "input" => %{} = input}),
+    do: [{property, map_size(input)}]
+
+  def get_properties(%{"property" => property}), do: [{property, 1}]
+  def get_properties(_data), do: []
+
+  def print_property({property, num_variables}) do
+    variables = Enum.map(1..num_variables, fn n -> to_string([?a - 1 + n]) end)
+
+    """
+     @doc \"\"\"
+     insert function description here
+     \"\"\"
+     @spec #{property}(#{Enum.map_join(variables, ", ", fn var -> "#{var} :: #{var}_type()" end)}) :: {:ok, return_type()} | {:error, String.t()}
+     def #{property}(#{Enum.join(variables, ", ")}) do
+     end
+    """
+  end
+
+  def print_comments(comments, do_not_print) do
+    comments
+    |> Enum.reject(fn {field, _value} -> field in do_not_print end)
+    |> Enum.map_join("\n", fn
+      {"comments", values} when is_list(values) ->
+        "# #{Enum.join(values, "\n# ")}"
+
+      {field, values} when is_list(values) ->
+        "#\n# --#{field} -- \n# #{Enum.map_join(values, "\n# ", &inspect/1)}"
+
+      {field, value} ->
+        "#\n# -- #{field} --\n# #{inspect(value)}"
+    end)
+  end
+
+  def print_test_case(%{"description" => description, "cases" => sub_cases} = c, module) do
+    """
+    describe \"#{description}\" do
+    #{Generate.print_comments(c, ["description", "cases"])}
+    #{Enum.map_join(sub_cases, "\n\n", &Generate.print_test_case(&1, module))}
+    end
+    """
+  end
+
+  def print_test_case(
+        %{
+          "description" => description,
+          "property" => property,
+          "input" => %{} = input,
+          "expected" => expected
+        } = c,
```

Comment:
Good to know, thank you! As of now, it would print the test in the following way
```elixir
test "disallow empty first strand" do
# Normalises error messages
# -- reimplements --
# db92e77e-7c72-499d-8fe6-9354d2bfd504
...
```
Which is not that bad, since there needs to be a manual check anyway.